queries.sql

```diff
@@ -40,3 +40,61 @@ SELECT *
 FROM animals
 WHERE weight>=10.4 and weight<=17.3 
 
+/**ex-2*/
+/*Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.*/
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon'
+
+/*Update the animals table by setting the species column to pokemon for all animals that don't have species already set*/
+UPDATE animals
+SET species="pokemon"
+WHERE species IS NULL;
+
+/* delete all records in the animals table, then roll back the transaction.*/
+DELETE FROM animals
+
+/* Delete all animals born after Jan 1st, 2022.*/
+DELETE FROM animals
+WHERE date_of_birth > "01/01/2022"
+
+/*Update all animals' weight to be their weight multiplied by -1*/
+UPDATE animals
+SET weight=weight*(-1)
+
+/*Update all animals' weights that are negative to be their weight multiplied by -1*/
+UPDATE animals
+SET weight=weight*(-1)
+WHERE weight < 0 ;
```

Comment:
- [ ] You should put this Inside a transaction, Create a savepoint for the transaction, Rollback to the savepoint and commit, It may look like this  :point_down: 

```sql
BEGIN;

    DELETE FROM animals WHERE  date_of_birth >= '2022/01/01' ;
    SAVEPOINT SP1;

    UPDATE animals SET weight_kg = weight_kg * -1; 
    ROLLBACK TO SP21;
    UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 

    COMMIT;
```